### ISSUE TITLE
Problem with `markdown-in-html`

### ISSUE THREAD

**Issue by Explorare** [trentm/python-markdown2#185]
[Problem with `markdown-in-html` · Issue #108 · bordaigorl/sublime-evernote](https://github.com/bordaigorl/sublime-evernote/issues/108#issuecomment-137090888)

> the `<markdown2>` library fails to handle the innermost `<div>` correctly (for some reason it treats it as a paragraph instead of raw html).

```
<div markdown="1">

<div>

</div>

</div>
```

gets wrongly translated by `<markdown2>` to
    <div>

```
<div>


md5-4eb9d42dbe0d156e1b0bffc28b03212a



</div>
```

**Comment by fangzhzh** [comment]
Any update for this issue?

I found the same problem for https://github.com/bordaigorl/sublime-evernote.
For the code below, I got Attribute "markdown" must be declared for element type "div".
	Last command:  {}.
`<div markdown="1" style="margin:0px;padding:0px;box-sizing:border-box;clear:both;min-height:1em;">
</div>`

**Comment by Explorare** [comment]
Umm... I didn't use it for a while. And I tried to reappear this bug but failed. I replaced the `markdown2.py` in `sublime-evernote` with the newer version in this repo. I also found that you will meet the same error log if you don't close the `<div>` correctly with `</div>`. Hope I could help you.

**Comment by ceteri** [comment]
I did further testing with the latest (v2.4.9) release.

This also causes the issue where an MD5 checksum gets substituted in the output instead of a paragraph, when reading Markdown input which has HTML in it.

  * `markdown2` v2.4.9
  * Python v3.9.0:9cf6752276
  * macOS 12.6.3

For example, running this code fragment:

```python
from markdown2 import Markdown
MD = Markdown()

text = """<div
>
<h3>Archons of the Colophon</h3>
<p>by Paco Xander Nathan
</p>
</div>"""

print(MD.convert(text))
```

... that produces:

```
<div
>
<h3>Archons of the Colophon</h3>


md5-be2124dfcd2a5199f4f6052c63916ae3

</div>
```

Note the `md5-be2124dfcd2a5199f4f6052c63916ae3` value and the missing text.

However, changing the `text` variable in either of two ways:

  * place the opening `<div>` all on one line
  * remove the newline before the closing `</p>`

... such as in the following code:

```python
text = """<div>
<h3>Archons of the Colophon</h3>
<p>by Paco Xander Nathan
</p>
</div>"""
```

... now produces correct output:

```
<div>
<h3>Archons of the Colophon</h3>
<p>by Paco Xander Nathan
</p>
</div>
```

I also found that pinning the dependency to the previous release `markdown2==2.4.8` resolves the problem.

That said, this issue was reported 8 years ago and there have been many releases between then.

Looking into the code, this seems likely related to one of the calls to `_hash_text()` and [this section](https://github.com/trentm/python-markdown2/blob/863a3af631fc8f805a0002b08e209608a13d49a9/lib/markdown2.py#L1404) seems likely?

```python
            if is_html_markup and not _is_auto_link(token) and not _is_code_span(index, token):
                sanitized = self._sanitize_html(token)
                key = _hash_text(sanitized)
                self.html_spans[key] = sanitized
                tokens.append(key)
            else:
                tokens.append(self._encode_incomplete_tags(token))
```

**Comment by Crozzers** [comment]
Hi, submitted a fix for this in #521. Seems to solve all the cases here from my testing
